### PR TITLE
[Chat] Update inline image / file sharing data type names

### PR DESCRIPTION
**Pull request by emlynmac**
# What
Update naming and data layout for inline images and file attachments following internal review
ChatMessage now has `files` and `inlineImages`

# Why
Simplify the API and clarify usage

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -150,6 +150,9 @@ export interface AttachmentDownloadResult {
     blobUrl: string;
 }
 
+// @beta
+export type AttachmentMetadata = FileMetadata | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ InlineImageMetadata;
```
Comment (JoshuaLai) [inline packages/communication-react/review/beta/communication-react.api.md:154]: Optional: Fixing the camel casing `MetaData` everywhere

File: packages/communication-react/review/beta/communication-react.api.md [hunk 2]
```diff
@@ -2697,20 +2696,18 @@ export interface FileDownloadError {
...
-// @beta
-export interface FileSharingMetadata extends BaseFileMetadata {
+export interface FileMetadata {
     // (undocumented)
-    attachmentType: 'fileSharing';
+    attachmentType: 'file';
+    extension: string;
+    id: string;
+    name: string;
     // (undocumented)
```
Comment (Leah-Xia-Microsoft) [inline packages/communication-react/review/beta/communication-react.api.md:2708]: Optional: Would be nice to add document of what this payload is about and what are the keys and values.

**Comment by Leah-Xia-Microsoft** [comment]
In the storybook generated from this PR (https://60c7ae6891f0e90039d7cd54-zpwcqifwwg.chromatic.com/?path=/story/ui-components-messagethread--message-thread),
when sending a message with inline image in the Preview tab, the image is not displayed. Is it related to the changes in this PR?

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (emlynmac) [reply packages/communication-react/review/beta/communication-react.api.md:154]: `Metadata` is the correct form. It is one word and not two.

File: packages/communication-react/review/beta/communication-react.api.md (hunk 2 above)
Comment (emlynmac) [reply packages/communication-react/review/beta/communication-react.api.md:2708]: The type is documented: https://github.com/Azure/communication-ui-library/pull/3846/files/0c588a84d91b17d2c283c66bb7bc83abfc6a1bcb#diff-d257e524dc363b5145a6b1c8b21f6f13fa9238b80aa0e665ce5505edc58ee9d9R29
I'm not sure why it's not coming through at this level with the tooling,.

File: change/@azure-communication-react-a596f1ce-68e3-45d1-b493-96bff2bdaedf.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "type": "prerelease",
```
Comment (palatter) [inline change/@azure-communication-react-a596f1ce-68e3-45d1-b493-96bff2bdaedf.json:2]: Is type supposed to be prerelease?

File: packages/communication-react/review/beta/communication-react.api.md [hunk 4]
```diff
@@ -1649,7 +1646,7 @@ export type ChatHandlers = {
     onLoadPreviousChatMessages: (messagesToLoad: number) => Promise<boolean>;
     onUpdateMessage: (messageId: string, content: string, options?: {
         metadata?: Record<string, string>;
-        attachedFilesMetadata?: FileMetadata[];
+        attachedFilesMetadata?: AttachmentMetadata[];
```
Comment (palatter) [inline packages/communication-react/review/beta/communication-react.api.md:1649]: Should the variable name here include 'Files'?

**Review by vhuseinova-msft — APPROVED** [review]

File: packages/communication-react/review/beta/communication-react.api.md (hunk 4 above)
Comment (palatter) [reply packages/communication-react/review/beta/communication-react.api.md:1649]: in other places we just use metadata

image

Comment (palatter) [reply packages/communication-react/review/beta/communication-react.api.md:1649]: Would this only be for files or would inline image also have attachmentMetaData?

Comment (emlynmac) [reply packages/communication-react/review/beta/communication-react.api.md:1649]: We can make additional naming changes if we want to, are you suggesting we rename to `attachmentMetadata` ?

Comment (palatter) [reply packages/communication-react/review/beta/communication-react.api.md:1649]: That would be my suggestion if it does also cover inline image. 'attachmentMetadata' would be more clear. What do you think?

Comment (emlynmac) [reply packages/communication-react/review/beta/communication-react.api.md:1649]: I'll go through the interfaces and adjust :)

**Pull request merged**